### ISSUE TITLE
'no-unused-vars' false flag when variable only used in template string

### ISSUE THREAD

**Issue by scriptdaemon** [eslint/eslint#5408]
**What version of ESLint are you using?** 2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?** Defaults

**What did you do? Please include the actual source code causing the issue.**

The code in question:

    function sendSteamRepReport(chatId, userId) {
      let options = {
        'url'  : 'http://steamrep.com/api/beta/reputation/${userId}?json=1',
        'json' : true
      };
      request.get(options, (err, response, body) => {
        if (err) {
          console.error(err.toString());
          return;
        }

        if (body && body.steamrep) {
          let steamrep   = body.steamrep;
          let reputation = steamrep.reputation;
          if (steamrep.flags.status === 'exists' && reputation) {
            let name = getDisplayName(userId);
            let msg  = 'SteamRep report for ${name}: ${reputation}';
            steamFriends.sendMessage(chatId, msg);
          }
        }
      });
    }

`name` is declared the line prior to and used only in the line `let msg  = 'SteamRep report for ${name}: ${reputation}';`.

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

`error  'name' is defined but never used  no-unused-vars`

**Comment by BYK** [comment]
Well, looking closely `let msg  = 'SteamRep report for ${name}: ${reputation}';` is not template literals :)

**Comment by BYK** [comment]
Also I verified that the following works on master:
```js
let who = "Paul";
let text = `Hello ${who}!`;
console.log(text);
```

**Comment by ilyavolodin** [comment]
@scriptdaemon template literals will only work if you use \` quotes around them. Otherwise those are just strings.
